.github/actions/notary-service/action.yaml

```diff
@@ -0,0 +1,41 @@
+name: notary-service
+description: 'Sets up a notary service (signer or server)'
+inputs:
+  service:
+    description: "Which kind of service is to be started (signer or server)"
+    required: true
+  port_map:
+    description: "The port mapping for the service container"
+    required: true
+  add_host:
+    description: "What additional hosts to add to the service"
+    required: false
+outputs:
+  ip:
+    description: "IP address of the notary service"
+    value: ${{ steps.get_service_ip.outputs.service_ip }}
+runs:
+  using: composite
+  steps:
+    - name: Setup service
+      run: |
+        DOCKER_ARGS='-d -p ${{ inputs.port_map }} -v ./test/integration/self-hosted-notary/notary-service-container/${{ inputs.service }}:/etc/docker/notary-${{ inputs.service }}'
+        SERVICE_ARGS='-config=/etc/docker/notary-${{ inputs.service }}/config.json'
+
+        if [[ "${{ inputs.service }}" == "server" ]]; then
```

Comment:
Would be more general, if you checked for add_host being non-empty and then add it